libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -695,6 +696,8 @@ uint32_t AP_Frsky_Telem::calc_ap_status(void)
     ap_status |= (uint8_t)(AP_Notify::flags.failsafe_battery)<<AP_BATT_FS_OFFSET;
     // bad ekf flag
     ap_status |= (uint8_t)(AP_Notify::flags.ekf_bad)<<AP_EKF_FS_OFFSET;
+    // IMU temperature: offset -19, 0 means temp =< 19°, 63 means temp => 82°
+    ap_status |= ((imu_temp < AP_IMU_TEMP_ORIGIN) ? 0 : (imu_temp > (AP_IMU_TEMP_ORIGIN + AP_IMU_TEMP_LIMIT)) ? AP_IMU_TEMP_LIMIT : (uint8_t)roundf(imu_temp - AP_IMU_TEMP_ORIGIN))<<AP_IMU_TEMP_OFFSET;
```

Comment:
please do this as multiple if statements to make it more readable

Reply:
This reads as:
if (imu_temp < AP_IMU_TEMP_ORIGIN)
    ap_status |= 0;
else if (imu_temp > (AP_IMU_TEMP_ORIGIN + AP_IMU_TEMP_LIMIT))
    ap_status |= AP_IMU_TEMP_LIMIT;
else ap_status |= (uint8_t)roundf(imu_temp - AP_IMU_TEMP_ORIGIN)<<AP_IMU_TEMP_OFFSET;

This may not be 100% correct so please check...

Reply:
thank you @floaledm i have read your code simplification to late, have done nearly the same way...